src/reflect/type.go

```diff
@@ -684,7 +684,13 @@ func (t rawType) NumMethod() int {
 }
 
 func (t rawType) Name() string {
-	panic("unimplemented: (reflect.Type).Name()")
+	kind := t.Kind()
+	switch kind {
+	case Bool, Int, Int8, Int16, Int32, Int64, Uint, Uint8, Uint16, Uint32, Uint64, Float32, Float64, Complex64, Complex128, String:
+		return kind.String()
+	default:
+		return ""
```

Comment:
```suggestion
		panic("unimplemented for named types: (reflect.Type).Name()")
```

Reply:
FYI, we had a discussion in this [thread on slack](https://gophers.slack.com/archives/CN4R2DQV7/p1657881610202449) with @aykevl. There is one [PR refactoring the reflect package](https://github.com/tinygo-org/tinygo/pull/2640) that is in the waiting room :)  So it will change after the PR is merged... I am unsure about the delays concerning the merge of the PR.